### ISSUE TITLE
Is the Central Analyzer required to avoid False Positives?

### ISSUE THREAD

**Issue by MichaelVetter** [dependency-check/DependencyCheck#2242]
We use the maven plugin to check complete archives (war and ear) without the dependency information from pom.xml files. When the Central Analyzer is disabled (default setting), we get several False Positives for Spring libraries (e.g. Spring Webflow, Spring Webservice, etc.).
Should we report this as False Positives or do you consider the Central Analyzer as required in such a scenario?

**Comment by jeremylong** [comment]
Can you tell me which version of dependency-check you are using and which spring libraries you see false positives on if the Central Analyzer is disabled?

**Comment by MichaelVetter** [comment]
The problem occurs using version 5.2.2 of the maven plugin.
We have false positives with these libraries:
````xml
<filePath regex="true">.*\spring-faces.*\.jar</filePath>
<filePath regex="true">.*\spring-binding.*\.jar</filePath>
<filePath regex="true">.*\spring-ws-.*\.jar</filePath>
<filePath regex="true">.*\spring-xml.*\.jar</filePath>
<filePath regex="true">.*\spring-security-saml.*\.jar</filePath>
````

**Comment by jeremylong** [comment]
Can you elaborate on the FP?  What version of the libraries are you using?  I.e. can you provide a snippet from the `<dependencies>` section instead of snippets from the suppression file you've generated.

**Comment by MichaelVetter** [comment]
The dependency checker generates CPEs of spring framework and spring security for these libraries which is wrong because they have their own releases with different version numbers:

```xml
                  <groupId>org.springframework.webflow</groupId>
                  <artifactId>spring-faces</artifactId>
                  <groupId>org.springframework.webflow</groupId>
                  <artifactId>spring-binding</artifactId>
		  <version>2.4.6.RELEASE</version>

                  <groupId>org.springframework.ws</groupId>
                  <artifactId>spring-ws-core</artifactId>
                  <groupId>org.springframework.ws</groupId>
                  <artifactId>spring-xml</artifactId>
		  <version>2.2.1.RELEASE</version>

      <groupId>org.springframework.security.extensions</groupId>
      <artifactId>spring-security-saml2-core</artifactId>
      <version>1.0.9.RELEASE</version>
```

**Comment by jeremylong** [comment]
I was able to test this today and while I agree that there are FP (which will be taken care of shortly) I found no difference between having the Central Analyzer disabled or enabled when using the Maven plugin.

**Comment by jeremylong** [comment]
Wait - I just re-read what you wrote above.  How are you using the Maven plugin to test EAR/WAR files?  If you are using the `scanSets` to scan JAR files (or ear/war files) then yes - the Central Analyzer would need to be enabled.  This is NOT a recommended use case. I would highly recommend scanning the build that is producing the EAR/WAR files as opposed to trying to scan them after the fact.

**Comment by MichaelVetter** [comment]
We do this using scanSets to make sure we only scan what is in the final artifact when modifications of the normal maven dependencies (like assembly plugin) are used.

**Comment by jeremylong** [comment]
The intent of the scanSets was to ensure that JavaScript files are scanned. In your use case - enable the Central Analyzer.

**Comment by MichaelVetter** [comment]
I do not understand the content of the linked commit. Why do you suppress these CPEs?
- spring-faces and spring-js are part of spring webflow 
- spring-aop and spring-oxm are part of spring framework

**Comment by jeremylong** [comment]
Aren't sprint-aop and spring-oxm versioned differently than spring framework?  Same for spring-faces and spring-js in comparison to spring-webflow?

**Comment by MichaelVetter** [comment]
No, they are part of the same git repositories:
https://github.com/spring-projects/spring-framework
https://github.com/spring-projects/spring-webflow

**Comment by jeremylong** [comment]
Got it - thanks.  I reverted the changes for those.
